src/components/HomePage/RepositoryList/RepositoryListItem/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useCallback, useState } from "react";
+import { RepositoryListItemProps } from "../../../../interfaces";
+import { IoHeartOutline } from "react-icons/io5";
+import { IoHeart } from "react-icons/io5";
+import "./RepositoryListItem.css";
+import { useDispatch } from "react-redux";
+import { addRepository, removeRepository } from "../../../../redux";
+
+const RepositoryListItem = ({ repository }: RepositoryListItemProps) => {
+  const [favorite, setFavorite] = useState(false);
+  const dispatch = useDispatch();
+
+  const favChangeHandler = useCallback(() => {
+    const curFavorite = favorite;
+    setFavorite((prevState) => {
+      return !prevState;
+    });
+    if (curFavorite) {
+      dispatch(removeRepository(repository));
+    } else {
+      dispatch(addRepository(repository));
+    }
```

Comment:
Make your code more readable, do this throughout your repository 
```suggestion 
    if (favorite) {
      dispatch(removeRepository(repository));
    } else {
      dispatch(addRepository(repository));
    }
    
    setFavorite(prevState => !prevState);
```